src/editor/WelcomeDialog.h

```diff
@@ -0,0 +1,48 @@
+﻿#ifndef WELCOMEDIALOG_H
+#define WELCOMEDIALOG_H
+
+#include <QDialog>
+#include <QLabel>
+#include <QToolTip>
+#include <QProxyStyle>
+
+namespace Ui {
+  class WelcomeDialog;
+}
+
+class WelcomeDialog : public QDialog
+{
+  Q_OBJECT
+
+public:
+  explicit WelcomeDialog(QWidget *parent = nullptr);
+  ~WelcomeDialog();
+
+signals:
+  void newSelected();
+  void openSelected();
+  void connectSelected();
+
+protected slots:
+  void closeEvent(QCloseEvent *) override;
+  void buttonNewPressed()
```

Comment:
nit, would you mind defining these slots in the cpp file to keep implementations in one place?

Reply:
wdym